aspnetcore/native-aot/index.md

```diff
@@ -0,0 +1,183 @@
+{
+
+}
+```
+
+Because unused code is trimmed during publishing for Native AOT, the application cannot use unbounded reflection at runtime. Source generators are used to produce code to avoid the need for reflection. In some cases source generators produce code optimized for AOT even when a generator is not strictly required. To view source code that is generated based on the code in ```Program.cs``` modify the ```MyFirstAotWebApi.csproj``` to include the ```<EmitCompilerGeneratedFiles>true</EmitCompilerGeneratedFiles>``` property. Example:
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk.Web">
+
+  <PropertyGroup>
+    <!-- Other properties ommitted for brevity -->
+    <EmitCompilerGeneratedFiles>true</EmitCompilerGeneratedFiles>
+  </PropertyGroup>
+
+</Project>
+```
+
+With the project file updated, run the `dotnet build` command (`publish` isn't necessary to view generated code). In the built output file there will be a `obj/Debug/net8.0/generated/` directory which contains all the generated files for the project.
+
+When `dotnet publish` is run, the project source files and generated source files are compiled as normal and then outputted assemblies are passed into an native IL compiler which produces the native executable which contains the native machine code to run the application.
+
+
+## Benefits of using .NET native AOT with ASP.NET Core
+
+Using the .NET native AOT deployment model provides the following benefits:
+
+* **Minimize disk footprint**: When publishing using native AOT a single executable is produced containing just the code from external dependencies that is used to support the program. Reduced executable size can lead to:
+  * Smaller container images, for example in containerized deployment scenarios.
+  * Reduce deployment time from smaller images.
+* **Reduced startup time**: Native AOT applications can show reduced start-up times. Reduced start-up means:
+  * The app is ready to service requests quicker.
+  * Improved deployment where container orchestrators need manage transition from one version of the app to another.
+* **Reduce memory demand**: Native AOT apps can have reduced memory demands depending on the work being performed by the app. Reduced memory consumption can lead to greater deployment density and improved scalability.
+
+We ran the template application in our benchmarking lab to see what the differences were in terms of application size, memory use, and CPU and observed the following results:
+
+![Chart showing comparison of application size, memory use and startup time metrics of an AOT published app, a runtime app that is trimmed, and an untrimmed runtime app.](_static/aot-runtime-trimmed-perf-chart.png)
+
+You can see that native AOT has a dramatically lower application size on disk and memory utilization is also lower for our template scenario. Startup time is also significantly reduced. 🚀
+
+## ASP.NET Core and Native AOT Supportability
```

Comment:
```suggestion
## ASP.NET Core and native AOT compatibility
```
Supportability is apparently a real word - https://www.google.com/search?q=supportability - but compatibility feels like it makes more sense to me.